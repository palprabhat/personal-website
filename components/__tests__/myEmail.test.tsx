import { render } from "@testing-library/react";
import { myEmail as myEmailId } from "../../constants/testIds.const";
import MyEmail from "../myEmail";

describe("MyEmail Component", () => {
  it("renders correct content", () => {
    const { container, queryByTestId } = render(<MyEmail />);
    const email = queryByTestId(myEmailId);

    expect(email).toBeInTheDocument();
    expect(email).toHaveTextContent("prabhatpal.14@gmail.com");

    expect(container).toMatchSnapshot();
  });
});
