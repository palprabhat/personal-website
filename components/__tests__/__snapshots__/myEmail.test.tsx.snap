// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyEmail Component renders correct content 1`] = `
<div>
  <a
    class="flex justify-center items-center px-2 group"
    data-testid="my-email"
    href="mailto:prabhatpal.14@gmail.com"
  >
    <span
      class="brand-icon-email mr-1 group-hover:text-primary-600 group-focus:text-primary-600"
    />
    prabhatpal.14@gmail.com
  </a>
</div>
`;
